### PR TITLE
Allow numbers in descendant-combined type selectors (fix #1836)

### LINKED ISSUE #1836: [CSS] Descendant type selectors can't have a numeric in their name
Consider the following code:

```python
from textual.app        import App, ComposeResult
from textual.containers import Vertical
from textual.widgets    import Header, Footer, Label

class LabelH1( Label ):
    ...

class CSSOddnessApp( App[ None ] ):

    CSS = """
    Vertical LabelH1 {
        background: red;
    }
    """

    def compose( self ) -> ComposeResult:
        yield Header()
        yield Vertical(
            Label( "Label" ),
            LabelH1( "LabelH1" ),
        )
        yield Footer()

if __name__ == "__main__":
    CSSOddnessApp().run()
```

When run we get the following error:

```
 Error in stylesheet:
 /Users/davep/develop/python/textual-sandbox/css_oddness.py:CSSOddnessApp:1:19
╭────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ ❱ 1 │                                                                                                                                                      │
│   2 │   Vertical LabelH1 {                                                                                                                                 │
│   3 │   │   background: red;                                                                                                                               │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
 • Expected one of 'combinator child', 'comment start', 'declaration set start', 'new selector', 'pseudo class', 'selector', 'selector class', 'selector id',
 'selector universal', or 'whitespace'.
 • Did you forget a semicolon at the end of a line?
```

The same thing happens with `Vertical LabelH1`. On the other hand, if I remove the number from the inherited label widget:

```python
from textual.app        import App, ComposeResult
from textual.containers import Vertical
from textual.widgets    import Header, Footer, Label

class LabelHOne( Label ):
    ...

class CSSOddnessApp( App[ None ] ):

    CSS = """
    Vertical LabelHOne {
        background: red;
    }
    """

    def compose( self ) -> ComposeResult:
        yield Header()
        yield Vertical(
            Label( "Label" ),
            LabelHOne( "LabelHOne" ),
        )
        yield Footer()

if __name__ == "__main__":
    CSSOddnessApp().run()
```

this works fine. Likewise, if I retain the name but *don't* use combination:

```python
from textual.app        import App, ComposeResult
from textual.containers import Vertical
from textual.widgets    import Header, Footer, Label

class LabelH1( Label ):
    ...

class CSSOddnessApp( App[ None ] ):

    CSS = """
    LabelH1 {
        background: red;
    }
    """

    def compose( self ) -> ComposeResult:
        yield Header()
        yield Vertical(
            Label( "Label" ),
            LabelH1( "LabelH1" ),
        )
        yield Footer()

if __name__ == "__main__":
    CSSOddnessApp().run()
```

that also works fine.

I would suspect a variation on #1253.

### PR DESCRIPTION
**Pull request by davep**
This fixes #1836; it'll also help folk style a `Markdown` with a bit more specificity too.

*NOTE:* Looks like we may need some tests around this sort of thing; I would have added one for this problem as part of this PR but it wasn't clear to me where in the tests it should go, or indeed if we have any around this right now. We may want to review this as part of this or as a followup.

### CODE REVIEW

**Comment by willmcgugan** [comment]
A reasonable test might be to have some CSS that we know should parse, and see if it parses without errors.

**Comment by davep** [comment]
> A reasonable test might be to have some CSS that we know should parse, and see if it parses without errors.

@willmcgugan Is there a spot in the tests for that sort of thing right now? I went looking, wanting to fit this in, but couldn't obviously see anything. If not I'll kick some testing off based around this.

**Comment by willmcgugan** [comment]
Don't believe so. Worth adding I think. Just one big string with known rules and edge cases.

**Comment by davep** [comment]
Cool. Sounds worthwhile. Will do.

**Comment by davep** [comment]
Actually, on second look, it looks like `tests/css/test_parse.py` was the obvious place to add some tests.

File: tests/css/test_parse.py [hunk 1]
```diff
@@ -1189,3 +1189,40 @@ def test_text_align_empty_uses_default(self):
...
+    def test_type_starts_with_number(self):
+        stylesheet = Stylesheet()
+        stylesheet.add_source("1TestType {}")
+        with pytest.raises(TokenError):
+            stylesheet.parse()
+
+    def test_combined_type_no_number(self):
+        for seperator in " >,":
+            stylesheet = Stylesheet()
+            stylesheet.add_source(f"StartType {seperator} TestType {{}}")
```
Comment (rodrigogiraoserrao) [inline tests/css/test_parse.py:1214]:
```suggestion
        for separator in " >,":
            stylesheet = Stylesheet()
            stylesheet.add_source(f"StartType {separator} TestType {{}}")
```

File: tests/css/test_parse.py [hunk 2]
```diff
@@ -1189,3 +1189,40 @@ def test_text_align_empty_uses_default(self):
...
+            assert len(stylesheet.rules) == 1
+
+    def test_combined_type_with_number(self):
+        for seperator in " >,":
+            stylesheet = Stylesheet()
+            stylesheet.add_source(f"StartType {seperator} TestType1 {{}}")
```
Comment (rodrigogiraoserrao) [inline tests/css/test_parse.py:1220]:
```suggestion
        for separator in " >,":
            stylesheet = Stylesheet()
            stylesheet.add_source(f"StartType {separator} TestType1 {{}}")
```

File: tests/css/test_parse.py [hunk 3]
```diff
@@ -1189,3 +1189,40 @@ def test_text_align_empty_uses_default(self):
...
+            assert len(stylesheet.rules) == 1
+
+    def test_combined_type_starts_with_number(self):
+        for seperator in " >,":
+            stylesheet = Stylesheet()
+            stylesheet.add_source(f"StartType {seperator} 1TestType {{}}")
```
Comment (rodrigogiraoserrao) [inline tests/css/test_parse.py:1226]:
```suggestion
        for separator in " >,":
            stylesheet = Stylesheet()
            stylesheet.add_source(f"StartType {separator} 1TestType {{}}")
```

**Review by rodrigogiraoserrao — APPROVED** [review]
Other than the typo you can fix in a jiffy, looks good to me!

**Review by willmcgugan — APPROVED** [review]
